### PR TITLE
Feature/quiz step 4

### PR DESCRIPTION
**Pull request by flipkoehler**

### CODE REVIEW

File: pages/404.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <Header />
+      <h1>404 - Die gewünschte Film konnte leider nicht gefunden werden</h1>
+      <Image
+        src={"/images/john-travolta.gif"}
+        alt="good old John Travolta searching for Mia Wallace"
+        width={380}
+        height={200}
+      />
+      <p>...aber es wird mal wieder Zeit für Pulp Fiction, oder?!</p>
+      <StyledLink href="/movies/680">Gib mir Pulp Fiction!</StyledLink>
```
Comment (MerleMak) [inline pages/404.tsx:18]: 😂👌🏻

File: components/Quiz/quizLogic.js [hunk 2]
```diff
@@ -61,16 +61,29 @@ export default function MovieQuiz() {
...
+      .filter((movie) => movie.occasion && movie);
+
+    // Quiz Step 4: this filters through the array of Step 3 and reduces it by selected occasion
+    const checkedOccasion = updatedQuestionAnswer[3].answerOptions
+      .filter((answer) => answer.checked && answer)
+      .map((a) => a.value);
+
+    const filteredMoviesStep4 = filteredMoviesStep3.filter((movie) => {
+      return movie.occasion.some((occasion) =>
+        checkedOccasion.includes(occasion)
```
Comment (MerleMak) [inline components/Quiz/quizLogic.js:77]: 👌🏻

**Review by MerleMak — APPROVED** [review]
really nice Philip! Keep up the great work 🚀👍🏼

**Comment by flipkoehler** [comment]
thank you for the review @MerleMak 😊

**Pull request merged**